video/video_samples.rb

```diff
@@ -0,0 +1,236 @@
+      puts "Locations:"
+
+      label_annotation.locations.each do |location|
+        if location.level == :VIDEO_LEVEL
+          puts "Entire video"
+        else
+          segment          = location.segment
+          start_in_seconds = segment.start_time_offset / 1000000.0
+          end_in_seconds   = segment.end_time_offset / 1000000.0
+
+          puts "#{start_in_seconds} through #{end_in_seconds}"
+        end
+      end
+    end
+  end
+
+  puts "Processing video for label annotations:"
+  operation.wait_until_done!
+  # [END analyze_labels_local]
+end
+
+def analyze_faces path:
+  # [START analyze_faces]
+  # path = "Path to a video file on Google Cloud Storage: gs://bucket/video.mp4"
+
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features     = [:FACE_DETECTION]
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video path, features do |operation|
+    raise operation.results.message? if operation.error?
+    puts "Finished processing."
+
+    # first result is retrieved because a single video was processed
+    annotation_result = operation.results.annotation_results.first
+
+    annotation_result.face_annotations.each do |face_annotation|
+      segment_counter = 1
+      puts "Thumbnail size: #{face_annotation.thumbnail.length}"
+
+      face_annotation.segments.each do |segment|
+        if segment.start_time_offset == -1 && segment.end_time_offset == -1
+          puts "Location #{segment_counter}: Entire video"
+        else
+          start_in_seconds = segment.start_time_offset / 1000000.0
+          end_in_seconds   = segment.end_time_offset / 1000000.0
+
+          puts "Location #{segment_counter}: #{start_in_seconds} through #{end_in_seconds}"
+        end
+
+        segment_counter += 1
+      end
+    end
+  end
+
+  puts "Processing video for face annotations:"
+  operation.wait_until_done!
+  # [END analyze_faces]
+end
+
+def analyze_safe_search path:
+  # [START analyze_safe_search]
+  # path = "Path to a video file on Google Cloud Storage: gs://bucket/video.mp4"
+
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features     = [:SAFE_SEARCH_DETECTION]
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video path, features do |operation|
+    raise operation.results.message? if operation.error?
+    puts "Finished processing."
+
+    # first result is retrieved because a single video was processed
+    annotation_result = operation.results.annotation_results.first
+
+    annotation_result.safe_search_annotations.each do |safe_search_annotation|
+      time_in_seconds = safe_search_annotation.time_offset / 1000000.0
+
+      puts "Time: #{time_in_seconds}"
+      puts "\tadult:   #{safe_search_annotation.adult}"
```

Comment:
Who is tracy?

Hard to read the words 😕 

```ruby
puts "\tadult:   #{safe_search_annotation.adult}"
puts "\tspoof:   #{safe_search_annotation.spoof}"
puts "\tmedical: #{safe_search_annotation.medical}"
puts "\tracy:    #{safe_search_annotation.racy}"
puts "\tviolent: #{safe_search_annotation.violent}"
```

Less challenging to read the words 😄 

```ruby
puts "adult:   #{safe_search_annotation.adult}"
puts "spoof:   #{safe_search_annotation.spoof}"
puts "medical: #{safe_search_annotation.medical}"
puts "racy:    #{safe_search_annotation.racy}"
puts "violent: #{safe_search_annotation.violent}"
```